tests/ui/borrowck/trait-impl-argument-difference-ice.rs

```diff
@@ -0,0 +1,24 @@
+trait MemoryUnit {
```

Comment:
Could you also add a issue number for this UI test?

```suggestion
// issue: rust-lang/rust#123414
trait MemoryUnit {
```